Product/Production/Services/DocumentQueryCore/src/main/java/gov/hhs/fha/nhinc/docquery/audit/transform/DocQueryAuditTransforms.java

```diff
@@ -182,20 +186,19 @@ private ParticipantObjectIdentificationType createQueryParticipantObjectIdentifi
             DocQueryAuditTransformsConstants.PARTICIPANT_QUERY_OBJ_ID_TYPE_CODE_SYSTEM,
             DocQueryAuditTransformsConstants.PARTICIPANT_QUERY_OBJ_ID_TYPE_DISPLAY_NAME);
 
-        if (queryId != null && !queryId.isEmpty()) {
+        if (NullChecker.isNotNullish(queryId)) {
             participantObject.setParticipantObjectID(queryId);
         }
         TypeValuePairType encoding = new TypeValuePairType();
         encoding.setType(DocQueryAuditTransformsConstants.QUERY_ENCODING_TYPE);
         encoding.setValue(DocQueryAuditTransformsConstants.UTF_8.getBytes());
         participantObject.getParticipantObjectDetail().add(encoding);
-        //TODO is the homeCommunityId of responding gateway ???
-        TypeValuePairType homeCommunityTypeValue = new TypeValuePairType();
-        homeCommunityTypeValue.setType(DocQueryAuditTransformsConstants.HOME_COMMUNITY_ID);
-        homeCommunityTypeValue.setValue(HomeCommunityMap.formatHomeCommunityId(
-            HomeCommunityMap.getLocalHomeCommunityId()).getBytes());
-        participantObject.getParticipantObjectDetail().add(homeCommunityTypeValue);
-
+        if (NullChecker.isNotNullish(hcid)) {
+            TypeValuePairType homeCommunityTypeValue = new TypeValuePairType();
+            homeCommunityTypeValue.setType(DocQueryAuditTransformsConstants.HOME_COMMUNITY_ID);
+            homeCommunityTypeValue.setValue(hcid.getBytes());
+            participantObject.getParticipantObjectDetail().add(homeCommunityTypeValue);
+        }
```

Comment:
Can  you refactor setting values for ParticipantObject TypeValuePairType into a separate method by passing the arguments to the method?